static/js/components/SourceSearchBar.jsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState, useRef } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import makeStyles from "@mui/styles/makeStyles";
+
+import TextField from "@mui/material/TextField";
+import Autocomplete from "@mui/material/Autocomplete";
+import SearchIcon from "@mui/icons-material/Search";
+import InputAdornment from "@mui/material/InputAdornment";
+import CircularProgress from "@mui/material/CircularProgress";
+
+import { GET } from "../API";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    "& .MuiOutlinedInput-root": {
+      "& fieldset": {
+        borderColor: "black",
+      },
+      "&:hover fieldset": {
+        borderColor: "black",
+      },
+      "&.Mui-focused fieldset": {
+        borderColor: "black",
+      },
+    },
+  },
+  textField: {
+    color: "black",
+  },
+  icon: {
+    color: "black",
+  },
+  paper: {
+    backgroundColor: "#fff",
+  },
+  // These rules help keep the progress wheel centered. Taken from the first example: https://material-ui.com/components/progress/
+  progress: {
+    display: "flex",
+    // The below color rule is not for the progress container, but for CircularProgress. This component only accepts 'primary', 'secondary', or 'inherit'.
+    color: theme.palette.info.main,
+    "& > * + *": {
+      marginLeft: theme.spacing(2),
+    },
+  },
+}));
+
+function useDebouncer(value, delay) {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+  return debouncedValue;
+}
+
+const SourceSearchBar = () => {
+  const dispatch = useDispatch();
+  const [options, setOptions] = useState([]);
+  const [value, setValue] = useState(null);
+  const [inputValue, setInputValue] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [open, setOpen] = useState(false);
+  const navigate = useNavigate();
+
+  const classes = useStyles();
+
+  const debouncedInputValue = useDebouncer(inputValue, 500);
+  const cache = useRef({});
+
+  useEffect(() => {
+    const get = (val) =>
+      dispatch(
+        GET(
+          `/api/sources?sourceID=${val}&pageNumber=1&totalMatches=25&includeComments=false&removeNested=true`,
+          "skyportal/FETCH_AUTOCOMPLETE_SOURCES"
```

Comment:
We usually don't call these pages directly, but maybe there's a reason here that I don't understand. Besides, I'm not even sure when this gets called, since it looks like when you choose a source it loads the individual source page, and not the table with these parameters (as it should!).

Reply:
Thank you for your feedback ! 
To be honest, I reused the code of the QuickSearchBar component (that has the same utility) and adapted it to the sources page. So I'm not really sure of what this call does, but if you consider it pointless, I can delete it.

Reply:
I tested the search bar without the useEffect get but the search bar didn't return any suggestions anymore... I guess this is the call that returns the matching results of the search, so it is indeed necessary. Do I leave it this way ? Apart from that, my last commit removed the clearOnBlur option !

Reply:
I guess that means it is necessary!